suite/tests/CMakeLists.txt

```diff
@@ -1290,6 +1290,13 @@ if (CMAKE_CROSSCOMPILING AND DEFINED CMAKE_FIND_ROOT_PATH AND NOT DR_COPY_TO_DEV
   set(dr_test_ops -xarch_root ${CMAKE_FIND_ROOT_PATH} ${dr_test_ops})
 endif ()
 
+if (APPLE AND AARCH64)
+  # On macOS + arm64 tests will randomly fail depending on
+  # whether mmap returns a region satisfying 32-bit reachability constraints.
+  # Reducing the vm_size appears to eliminate this issue.
+  set(dr_test_ops -vm_size 500M ${dr_test_ops})
```

Comment:
https://github.com/DynamoRIO/dynamorio/issues/7296#issuecomment-2679125060

As pointed out by Derek, this shouldn't be the cause for a fatal error.
What happens if you use `-no_vm_base_near_app`?

Reply:
It turns out you can avoid the assert with `-no_reachable_client`. Is that a better temporary solution here? Presumably we eventually should fix #7296 so we can allocate >1GB near clients... 

https://github.com/DynamoRIO/dynamorio/issues/7296#issuecomment-2682457823

Reply:
See https://github.com/DynamoRIO/dynamorio/issues/7296#issuecomment-2682906760: we should turn off -reachable_client on Mac and document that.